run.py

```diff
@@ -56,6 +57,22 @@ def py_shell():
     shell.interact()
 
 
+def python_runtime():
+    # extract the absolute script path to alter sys.path and specific dunder variables
+    script = pathlib.Path(sys.argv[2]).expanduser().absolute()
+    sys.path.insert(0, str(script.parent))
```

Comment:
```suggestion
    script = pathlib.Path(sys.argv[2]).expanduser().resolve()
    sys.path.insert(0, str(script.parent))
```
`pathlib.Path.absolute()` is undocumented, so it might change on use, let's use `.resolve()`.

And from the `help()` call on a python shell
```
Help on function absolute in module pathlib:

absolute(self)
    Return an absolute version of this path.  This function works
    even if the path doesn't point to anything.

    No normalization is done, i.e. all '.' and '..' will be kept along.
    Use resolve() to get the canonical path to a file.
```
----
As much as I don't like that `sys.path.insert`, that's really what python does...

```
❯ pwd
/tmp
❯ python ~/tmp-proj/run-it.py
In __main__. 'sys.path': ['/home/vampas/tmp-proj', '/usr/lib/python310.zip', '/usr/lib/python3.10', '/usr/lib/python3.10/lib-dynload', '/usr/lib/python3.10/site-packages']
In foo. 'sys.path': ['/home/vampas/tmp-proj', '/usr/lib/python310.zip', '/usr/lib/python3.10', '/usr/lib/python3.10/lib-dynload', '/usr/lib/python3.10/site-packages']
In foo.bar. 'sys.path': ['/home/vampas/tmp-proj', '/usr/lib/python310.zip', '/usr/lib/python3.10', '/usr/lib/python3.10/lib-dynload', '/usr/lib/python3.10/site-packages']
```
```
❯ tree ~/tmp-proj
/home/vampas/tmp-proj
├── foo
│   ├── bar.py
│   └── __init__.py
└── run-it.py

1 directory, 3 file
```
```
❯ cat ~/tmp-proj/run-it.py
import sys
print("In {}. 'sys.path': {}".format(__name__, sys.path))

import foo
```
```
❯ cat ~/tmp-proj/foo/__init__.py
import sys

print("In {}. 'sys.path': {}".format(__name__, sys.path))

import foo.bar
```
```
❯ cat ~/tmp-proj/foo/bar.py
import sys
print("In {}. 'sys.path': {}".format(__name__, sys.path))
```